packages/next/client/head-manager.ts

```diff
@@ -44,15 +44,18 @@ function updateElements(type: string, components: JSX.Element[]) {
   const headCount = Number(headCountEl.content)
   const oldTags: Element[] = []
 
-  for (
-    let i = 0, j = headCountEl.previousElementSibling;
-    i < headCount;
-    i++, j = j!.previousElementSibling
-  ) {
-    if (j!.tagName.toLowerCase() === type) {
-      oldTags.push(j!)
+  let i = 0
+  let j = headCountEl.previousElementSibling
+  while (i < headCount) {
+    if (j!.getAttribute('data-next-head') === 'true') {
```

Comment:
Prefer safer optional chaining 
```suggestion
    if (j?.getAttribute('data-next-head') === 'true') {
```